client/my-sites/plugins/marketplace/marketplace-plugin-setup-status/index.tsx

```diff
@@ -1,40 +1,106 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { ThemeProvider } from 'emotion-theming';
 import { ProgressBar } from '@automattic/components';
 import styled from '@emotion/styled';
-import { useSelector } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
+import page from 'page';
+import { useTranslate } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import theme from 'calypso/my-sites/plugins/marketplace/theme';
 import Masterbar from 'calypso/layout/masterbar/masterbar';
-import { getStatusForPlugin } from 'calypso/state/plugins/installed/selectors';
 import { getSelectedSite } from 'calypso/state/ui/selectors';
 import QueryJetpackPlugins from 'calypso/components/data/query-jetpack-plugins';
-
+import { initiateThemeTransfer } from 'calypso/state/themes/actions';
+import { getAutomatedTransferStatus } from 'calypso/state/automated-transfer/selectors';
 /**
  * Style dependencies
  */
 import 'calypso/my-sites/plugins/marketplace/marketplace-plugin-setup-status/style.scss';
+import { getPurchaseFlowState } from 'calypso/state/plugins/marketplace/selectors';
 
 const StyledProgressBar = styled( ProgressBar )`
 	margin: 20px 0px;
 `;
 
-function WrappedMarketplacePluginSetup( {
-	pluginSlug = 'wordpress-seo',
-}: {
-	pluginSlug?: string;
-} ): JSX.Element {
+// Stages
+
+function WrappedMarketplacePluginSetup(): JSX.Element {
+	const translate = useTranslate();
+
+	const STAGE_1 = translate( 'Installing plugin' );
+	const STAGE_2 = translate( 'Activating plugin' );
+	const stages = [ STAGE_1, STAGE_2 ];
+
+	const [ currentStage, setCurrentStage ] = useState( STAGE_1 );
+	const [ simulatedProgressPercentage, setSimulatedProgressPercentage ] = useState( 0 );
+
+	const interval = 2000;
+
+	const dispatch = useDispatch();
 	const selectedSite = useSelector( getSelectedSite );
+	const transferStatus = useSelector( ( state ) =>
+		getAutomatedTransferStatus( state, selectedSite?.ID )
+	);
+	const { pluginSlugToBeInstalled, isPluginInstalledDuringPurchase } = useSelector(
+		getPurchaseFlowState
+	);
+
+	useEffect( () => {
+		if ( pluginSlugToBeInstalled ) {
+			dispatch( initiateThemeTransfer( selectedSite.ID, null, pluginSlugToBeInstalled ) );
+		} else if ( isPluginInstalledDuringPurchase ) {
+			alert( 'To be implemented: polling wait to be implemented to check for transfer' );
```

Comment:
Actually this code will be replaced in follow up PRs. The alert is meant to inform internal users that there are unhandled cases without needing them to delve into the console log.
WDYT?

Reply:
I think that we shouldn't introduce alerts like this, even if it's in work-in-progress code.

Reply:
Any reason why not? 🤔

Reply:
I think it's a poor experience for internal testers, too 😉  Just because we're internal testers, that doesn't mean we can or have to get a poor user experience. 

There must be a better way to warn our internal testers that something is still in development than blocking the further execution of scripts while the alert is open. 

So the good reason here is: `alert()` blocks script execution. This may block anything that is currently in progress and lead to unexpected results. Internal testers may end up reporting bugs that don't really exist and are perceived solely because we're using `alert()`.

Reply:
Alert freezes script execution but then after we dismiss the alert everything resumes starts working as expected doesn't it? 🤔 . Of course I am not arguing that we should ship alerts to users but I have never experienced unexpected results due to the use of alerts.

In order for us to reach production we have got to fix these cases where I have used alerts. In that case I want this to deliberately pinch us without silently logging an error. I want this unimplemented path to be a relatively blocking experience since we have got to replace it before going to prod. 

Also if someone is expecting a behaviour during our early alpha testing even within the team. Having an alert is immensely useful since it readily alerts the team mates to the fact that its not yet implemented. Rather than just failing silently. Won't it be a better testing experience than failing silently, or failing by logging when we popup a script stopping alert to explain the user why this specific code path is not working. 🤔 

And since everything works as expected after alerting I saw it as a win win. 🙂

Reply:
I disagree here, but again, don't want to block this PR with my drive-by review. Feel free to proceed the best way you feel is fit for your project needs.